tfhe/src/high_level_api/backward_compatibility/keys.rs

```diff
@@ -28,66 +27,29 @@ impl Upgrade<ClientKey> for ClientKeyV0 {
     }
 }
 
-// This type was previously versioned using a manual implementation with a conversion
-// to a type where the inner key was name `integer_key`
-#[derive(Version)]
-pub struct ServerKeyV0 {
-    pub(crate) integer_key: Arc<IntegerServerKey>,
-}
-
-impl Upgrade<ServerKeyV1> for ServerKeyV0 {
-    type Error = Infallible;
-
-    fn upgrade(self) -> Result<ServerKeyV1, Self::Error> {
-        Ok(ServerKeyV1 {
-            key: self.integer_key,
-        })
-    }
-}
-
-#[derive(Version)]
-pub struct ServerKeyV1 {
-    pub(crate) key: Arc<IntegerServerKey>,
-}
-
-impl Upgrade<ServerKey> for ServerKeyV1 {
-    type Error = Infallible;
-
-    fn upgrade(self) -> Result<ServerKey, Self::Error> {
-        Ok(ServerKey {
-            key: self.key,
-            tag: Tag::default(),
-        })
-    }
+impl Deprecable for ServerKey {
+    const TYPE_NAME: &'static str = "ServerKey";
+    const MIN_SUPPORTED_APP_VERSION: &'static str = "TFHE-rs v0.10";
```

Comment:
Is this really needed ? Since the different key types are each independently versioned, I'm not sure if we need to deprecate the higher level ones.
The deprecation will be caught during the deserialization of the inner type.

Reply:
The same comment applies to all types that are not directly composed of a level container.

Reply:
I thought about it but the error message would be less clear in that case if we don't do a mass deprecation, let me know what you think we should do, worse case I can revert some of the changes, but as they are done now does not matter much normally and I'd rather avoid unnecessary changes _now_ to get the thing ready for the blockchain

Reply:
Yes makes sense, I think we can keep it like this